codec/dagcbor/roundtrip_test.go

```diff
@@ -0,0 +1,64 @@
+package dagcbor
+
+import (
+	"bytes"
+	"testing"
+
+	. "github.com/warpfork/go-wish"
+
+	"github.com/ipld/go-ipld-prime/fluent"
+	basicnode "github.com/ipld/go-ipld-prime/node/basic"
+)
+
+var n = fluent.MustBuildMap(basicnode.Style__Map{}, 4, func(na fluent.MapNodeAssembler) {
+	na.AssembleDirectly("plain").AssignString("olde string")
+	na.AssembleDirectly("map").CreateMap(2, func(na fluent.MapNodeAssembler) {
+		na.AssembleDirectly("one").AssignInt(1)
+		na.AssembleDirectly("two").AssignInt(2)
+	})
+	na.AssembleDirectly("list").CreateList(2, func(na fluent.ListNodeAssembler) {
+		na.AssembleValue().AssignString("three")
+		na.AssembleValue().AssignString("four")
+	})
+	na.AssembleDirectly("nested").CreateMap(1, func(na fluent.MapNodeAssembler) {
+		na.AssembleDirectly("deeper").CreateList(1, func(na fluent.ListNodeAssembler) {
+			na.AssembleValue().AssignString("things")
+		})
+	})
+})
+var serial = "\xa4eplainkolde stringcmap\xa2cone\x01ctwo\x02dlist\x82ethreedfourfnested\xa1fdeeper\x81fthings"
+
+func TestRoundtrip(t *testing.T) {
+	t.Run("encoding", func(t *testing.T) {
+		var buf bytes.Buffer
+		err := Encoder(n, &buf)
+		Require(t, err, ShouldEqual, nil)
+		Wish(t, buf.String(), ShouldEqual, serial)
+	})
+	t.Run("decoding", func(t *testing.T) {
+		buf := bytes.NewBufferString(serial)
+		nb := basicnode.Style__Map{}.NewBuilder()
+		err := Decoder(nb, buf)
+		Require(t, err, ShouldEqual, nil)
+		Wish(t, nb.Build(), ShouldEqual, n)
```

Comment:
Does `ShouldEqual` have access to enough of the internals to make a meaningful comparison between two `Node`s?